### ISSUE TITLE
IDE0060 suggests that the this parameter in an extension method is unused

### ISSUE THREAD

**Issue by tannergooding** [dotnet/roslyn#79621]
### Repro

Ensure that `IDE0060` is enabled and then observe the following code:
```csharp
using System;

public static class C
{
    public static void M(this string s)
    {
        Console.WriteLine("potato");
    }
}
```

### Expected Behavior

No diagnostic should be produced as this is an extension method and is intentionally exposed on the `T`, even if otherwise unused. This can be the case such as if the normal method would've been abstract and was the reason it was instance; but the extension method cannot be abstract and is just doing a best attempt to polyfill.

### Actual Behavior

The analyzer will end up reporting `warning IDE0060: Remove unused parameter 's'`.

**Comment by CyrusNajmabadi** [comment]
I believe this is by design as it definitely is calling out something strange. If the intent is to polyfill, and the arg truly is intended to be unused, then the recommended thing to do here is suppress the diagnostic. This then makes her code self documenting that not using the parameter is intentional and not a mistake.

**Comment by Youssef1313** [comment]
A future me will probably come and report the fix done here (if any) as false negative 😄 
I tend to think the current behavior is correct, and I agree with @CyrusNajmabadi that suppressing for this scenario is suitable.

**Comment by RenderMichael** [comment]
Roslynator has had similar proposals. It’s worth noting that this warning can be suppressed/addressed without a `#pragma`:

```C#
using System;

public static class C
{
    public static void M(this string s)
    {
        _ = s; // Polyfill for string.M() introduced in .NET 99
        Console.WriteLine("potato");
    }
}
```

**Comment by dodexahedron** [comment]
How about allowing use of the discard syntax for the `this` parameter, for when the method isn't going to actually access it?

```
public static void Foo( this Bar _, ... )
{
}
```
